Python scripts incompatible with Python 3

Python scripts seem not to be compatible with Python 3. On some computers, `#!/usr/bin/env python` returns Python 3. It happens, e.g., with Anaconda which sometimes prepends its directory to `PATH`.

``` shell
Scanning dependencies of target dox
Traceback (most recent call last):
  File "/home/karel/github/cmake-ext/tests/cmake-ext/seqan-prefix/src/seqan/dox/../util/bin/dox.py", line 14, in <module>
    sys.exit(main())
  File "/home/karel/github/cmake-ext/tests/cmake-ext/seqan-prefix/src/seqan/dox/../util/bin/dox.py", line 10, in main
    import seqan.dox.pure
  File "/home/karel/github/cmake-ext/tests/cmake-ext/seqan-prefix/src/seqan/util/py_lib/seqan/dox/pure.py", line 129
    print 'Scanning %s...' % path
                         ^
SyntaxError: Missing parentheses in call to 'print'
dox/CMakeFiles/dox.dir/build.make:57: recipe for target 'dox/CMakeFiles/dox' failed
make[5]: *** [dox/CMakeFiles/dox] Error 1
CMakeFiles/Makefile2:167: recipe for target 'dox/CMakeFiles/dox.dir/all' failed
make[4]: *** [dox/CMakeFiles/dox.dir/all] Error 2
Makefile:160: recipe for target 'all' failed
make[3]: *** [all] Error 2
CMakeFiles/seqan.dir/build.make:110: recipe for target 'cmake-ext/seqan-prefix/src/seqan-stamp/seqan-build' failed
make[2]: *** [cmake-ext/seqan-prefix/src/seqan-stamp/seqan-build] Error 2
CMakeFiles/Makefile2:215: recipe for target 'CMakeFiles/seqan.dir/all' failed
make[1]: *** [CMakeFiles/seqan.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2

```

Hello @cpockrandt, @h-2 and @marehr,

I think I have found the main problem. Seqan relies on CMake ``find_package (PythonInterp)
`` and does not check versions, see https://github.com/seqan/seqan/blob/3056fcaa07d902679c72fe6c149bcddccdf53ef8/dox/CMakeLists.txt#L12. So in my case, it finds Python 3 and tries to execute Pythons script with it:
```
On-y-go:dox karel$ make VERBOSE=1 2>&1 | tail -n 13
cd /Users/karel/github/cmake-ext/tests/cmake-ext/seqan-prefix/src/seqan-build/dox && /Users/karel/anaconda3/bin/python /Users/karel/github/cmake-ext/tests/cmake-ext/seqan-prefix/src/seqan/dox/../util/bin/dox.py -b /Users/karel/github/cmake-ext/tests/cmake-ext/seqan-prefix/src/seqan/dox/.. -i /Users/karel/github/cmake-ext/tests/cmake-ext/seqan-prefix/src/seqan/dox/../include/seqan -i /Users/karel/github/cmake-ext/tests/cmake-ext/seqan-prefix/src/seqan/dox/pages --image-dir /Users/karel/github/cmake-ext/tests/cmake-ext/seqan-prefix/src/seqan/dox/images
Traceback (most recent call last):
  File "/Users/karel/github/cmake-ext/tests/cmake-ext/seqan-prefix/src/seqan/dox/../util/bin/dox.py", line 14, in <module>
    sys.exit(main())
  File "/Users/karel/github/cmake-ext/tests/cmake-ext/seqan-prefix/src/seqan/dox/../util/bin/dox.py", line 10, in main
    import seqan.dox.pure
  File "/Users/karel/github/cmake-ext/tests/cmake-ext/seqan-prefix/src/seqan/util/py_lib/seqan/dox/pure.py", line 129
    print 'Scanning %s...' % path
                         ^
SyntaxError: Missing parentheses in call to 'print'
make[2]: *** [dox/CMakeFiles/dox] Error 1
make[1]: *** [dox/CMakeFiles/dox.dir/all] Error 2
make: *** [all] Error 2
```

This should help: https://cmake.org/cmake/help/v3.0/module/FindPythonInterp.html
> The Python_ADDITIONAL_VERSIONS variable can be used to specify a list of version numbers that should be taken into account when searching for Python. You need to set this variable before calling find_package(PythonInterp).

Btw. is it possible to skip the Dox step when building Seqan with ``-DSEQAN_BUILD_SYSTEM=SEQAN_RELEASE_LIBRARY``? Something like ``-DNO_DOX=1`` would be very useful: some compilation time would be saved and Python related problems would not appear.

--- a/dox/CMakeLists.txt
+++ b/dox/CMakeLists.txt
@@ -18,7 +18,6 @@ endif ()
 # find python 2.x
 if(NOT(${PYTHON_VERSION_MAJOR} MATCHES "2"))
     set(_PYTHON2_VERSIONS 2.7 2.6 2.5 2.4 2.3 2.2 2.1 2.0)
-    #set(_PYTHON2_VERSIONS i12.7 12.6 12.5 12.4 12.3 12.2 12.1 12.0)
     unset(_Python_NAMES)
     foreach(_CURRENT_VERSION IN LISTS _PYTHON2_VERSIONS)
         list(APPEND _Python_NAMES "python${_CURRENT_VERSION}")

--- a/dox/CMakeLists.txt
+++ b/dox/CMakeLists.txt
@@ -13,7 +13,26 @@ find_package (PythonInterp)
 if (NOT PYTHONINTERP_FOUND)
     message (STATUS "  Python not found, not building dox as a tests.")
     return ()
-endif (NOT PYTHONINTERP_FOUND)
+endif ()
+
+# find python 2.x
+if(NOT(${PYTHON_VERSION_MAJOR} MATCHES "2"))
+    set(_PYTHON2_VERSIONS 2.7 2.6 2.5 2.4 2.3 2.2 2.1 2.0)
+    #set(_PYTHON2_VERSIONS i12.7 12.6 12.5 12.4 12.3 12.2 12.1 12.0)
+    unset(_Python_NAMES)
+    foreach(_CURRENT_VERSION IN LISTS _PYTHON2_VERSIONS)
+        list(APPEND _Python_NAMES "python${_CURRENT_VERSION}")
+    endforeach()
+    unset(_PYTHON2_VERSIONS)
+
+    find_program(PYTHON2_EXECUTABLE NAMES ${_Python_NAMES})
+    if(PYTHON2_EXECUTABLE MATCHES "PYTHON2_EXECUTABLE-NOTFOUND")
+        message (STATUS "You need Python 2.x for building dox. (skip the tests)")
+        return()
+    endif()
+    set(PYTHON_EXECUTABLE "${PYTHON2_EXECUTABLE}")
+    message (STATUS "Found Python 2.x: ${PYTHON_EXECUTABLE}") 
+endif()
 
 # Add building the documentation as a test.
 add_test (build_dox

